### ISSUE TITLE
EnumerationGraphType#ParseLiteral recieves a StringValue instead of EnumValue

### ISSUE THREAD

**Issue by JarnoNijboer** [graphql-dotnet/graphql-dotnet#95]
The value param that EnumerationGraphType#ParseLiteral recieves is actually of type StringValue.
When it's cast to a EnumValue, the ```enumValue``` variable is null
![image](https://cloud.githubusercontent.com/assets/2635154/15925248/a884c7e6-2e36-11e6-90d5-cd6c0c243985.png)

This quick hack, resolves the problem temporaliy for me:
```csharp
public override object ParseLiteral(IValue value)
{
    var enumVal = value as StringValue;
    return ParseValue(enumVal?.Value);
}
```
It seems to me that EnumValues are processed as normal strings instead of as a value of an enum type

**Comment by joemcbride** [comment]
This has to be a particular query.  I'm guessing perhaps your query has the enum value sent as a string vs. just text.  `"Male"` vs. `Male`?

There is an example "enum as input objects" test here: https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.Tests/Execution/EnumAsInputsTests.cs

Are you able to verify it is getting sent to the parser correctly write a failing test?

**Comment by JarnoNijboer** [comment]
It seems that relay sends the enum value as a part of the input variable. The result of this is that the enum value is send as a string, and not as text.
At the moment I have not the time needed for creating a failing test, I will try to do this when I have the necessary time available.

An shortend example of the mutation that is send by relay:
```graphql
mutation AddPropertyMutation($input_0: AddPropertyInput!) {
  addProperty(input: $input_0) {
    clientMutationId
    ...F4
  }
}

fragment F4 on AddPropertyPayload {
  productType {
    id
    ...F0
  }
}
```
and the corresponding variables
```json
{
  "input_0": {
    "clientMutationId":"b",
    "type":"lookup"
  }
}
```

**Comment by tlil87** [comment]
The fact that the enum value is encoded as a string in the JSON input is correct. So I'm guessing that it's consequently not interpreted as an enum value following the translation from JSON to input object (which is not touched by the parser) @joemcbride. There doesn't seem to be an existing unit test for that scenario either (i.e. where it's passed in separately through the `variables` payload).

**Comment by joemcbride** [comment]
That would do it!  I'll add a test and a fix.

**Comment by joemcbride** [comment]
Fixed in [v0.8.2](https://github.com/graphql-dotnet/graphql-dotnet/releases/tag/v0.8.2)
